fgcm/fgcmGray.py

```diff
@@ -1003,17 +1003,18 @@ def makeExpGrayPlots(self):
 
         deepUse,=np.where(self.fgcmPars.expDeepFlag[expUse] == 1)
 
-        ax.hexbin(self.fgcmPars.expMJD[expUse[deepUse]] - firstMJD,
-                  expGray[expUse[deepUse]]*1000.0, bins='log')
-        ax.set_xlabel(r'$\mathrm{MJD}\ -\ %.0f$' % (firstMJD),fontsize=16)
-        ax.set_ylabel(r'$\mathrm{EXP}^{\mathrm{gray}}\,(\mathrm{mmag})$',fontsize=16)
+        if deepUse.size > 0:
+            ax.hexbin(self.fgcmPars.expMJD[expUse[deepUse]] - firstMJD,
+                      expGray[expUse[deepUse]]*1000.0, bins='log')
+            ax.set_xlabel(r'$\mathrm{MJD}\ -\ %.0f$' % (firstMJD),fontsize=16)
+            ax.set_ylabel(r'$\mathrm{EXP}^{\mathrm{gray}}\,(\mathrm{mmag})$',fontsize=16)
 
-        ax.set_title(r'$\mathrm{Deep Fields}$')
+            ax.set_title(r'$\mathrm{Deep Fields}$')
 
-        if self.plotPath is not None:
-            fig.savefig('%s/%s_mjd_deep_expgray.png' % (self.plotPath,
-                                                        self.outfileBaseWithCycle))
-        plt.close(fig)
+            if self.plotPath is not None:
+                fig.savefig('%s/%s_mjd_deep_expgray.png' % (self.plotPath,
+                                                            self.outfileBaseWithCycle))
+            plt.close(fig)
```

Comment:
Might it still be good to have this put.close(fig) trigger regardless (or perhaps move the figure opening into the `if deepUse.size > 0:` clause)?